NitroxServer/Communication/NetworkingLayer/NitroxServer.cs

```diff
@@ -67,5 +68,24 @@ protected void ProcessIncomingData(NitroxConnection connection, Packet packet)
                 Log.Error("Exception while processing packet: " + packet + " " + ex);
             }
         }
+
+        private void SetupUPNP()
+        {
+            NatUtility.DeviceFound += DeviceFound;
+            NatUtility.StartDiscovery();
+        }
+
+        private void DeviceFound(object sender, DeviceEventArgs args)
+        {
+            try
+            {
+                INatDevice device = args.Device;
+                device.CreatePortMapAsync(new Mapping(Protocol.Udp, portNumber, portNumber, 0, "Nitrox Server - Subnautica"));
```

Comment:
Does not say anything if succeeded. Probably best to let this throw an exception that should be handled so that the error message has more detail.

Reply:
Especially in debug mode I expect it to tell me if the port is already forwarded.

Reply:
Let's also auto expire so that the port doesn't remain open forever when they stop using Nitrox. A week or month sounds good to me.

Reply:
Added logs and auto expire of 2 hours, do we think that is long enough?
Also added debug logging when in debug mode and a more user friendly message for release build